### DISCUSSION TITLE
How to perform an assert.NotElementsMatch()

### DISCUSSION THREAD

**Discussion by jdycaico** [stretchr/testify#1592] [Q&A] [answered]
Hi,

Just getting started with Testify. Couldn't find a method to perform the above.
Context: 

> I want to assert that two slices of strings do NOT contain the same elements, irrespective of order.

How would I do this with Testify?

**Comment by brackendawson** [comment]
We should probably have a NotElementsMatch assertion, if you want to open an issue or pull request for that we would be open to adding it.

How would I do it right now? This is a hack, and it might not work in future Go versions:
```go
package kata_test

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIfy(t *testing.T) {
	s1 := []string{"cat", "bat", "rat"}
	s2 := []string{"bat", "cat", "rat"}
	s3 := []string{"cow", "pig", "dog"}
	assertNotElementsMatch(t, s1, s2)
	assertNotElementsMatch(t, s1, s3)
}

func assertNotElementsMatch(t *testing.T, listA, listB any) bool {
	t.Helper()
	return assert.False(t, assert.ElementsMatch(new(testing.T), listA, listB), "Lists have the same elements:\n%v\n%v", listA, listB)
}
```

**Reply by jdycaico** [reply]
Opened https://github.com/stretchr/testify/issues/1594
Thanks

**Comment by hendrywiranto** [comment, marked as answer]
this assertion has been added in https://github.com/stretchr/testify/pull/1600
I think we can close this question @jdycaico
